app/models/teacher.rb

```diff
@@ -0,0 +1,14 @@
+class Teacher < ApplicationRecord
+  include ActiveModel::Serializers::JSON
+
+  has_many :appointments, dependent: :destroy
+  has_many :users, through: :appointments
+  validates :name, presence: true
+
+  def attributes
```

Comment:
Ditto, this method isn't proper for a model

Reply:
This method has been removed